### PR TITLE
Let Cocoon write into metrics center

### PR DESCRIPTION
**Pull request by liyuqian**
This needs https://github.com/flutter/cocoon/pull/1061 and the `metrics_center` change in `pubspec.yaml` from https://github.com/flutter/cocoon/pull/1060.

Once this PR is successfully deployed, we should be able to cut the dependency on Tong's desktop.

### CODE REVIEW

**Comment by liyuqian** [comment]
@keyonghan : I think this is now ready for review as https://github.com/flutter/engine/pull/23767 has been successful on the engine side.

File: app_dart/lib/src/datastore/cocoon_config.dart [hunk 1]
```diff
@@ -333,6 +334,12 @@ class Config {
     return GithubService(github);
   }
 
+  Future<FlutterDestination> createMetricsDestination() async {
```
Comment (keyonghan) [inline app_dart/lib/src/datastore/cocoon_config.dart:337]: can we add a doc briefly explain Metrics/flutter destination?

File: app_dart/lib/src/request_handlers/update_task_status.dart [hunk 2]
```diff
@@ -122,6 +127,35 @@ class UpdateTaskStatus extends ApiRequestHandler<UpdateTaskStatusResponse> {
     return UpdateTaskStatusResponse(task);
   }
 
+  Future<void> _writeToMetricsCenter(
```
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/update_task_status.dart:130]: ditto.

**Review by keyonghan — APPROVED** [review]
LGTM with doc nits.

File: app_dart/lib/src/datastore/cocoon_config.dart (hunk 1 above)
Comment (liyuqian) [reply app_dart/lib/src/datastore/cocoon_config.dart:343]: Done.

File: app_dart/lib/src/request_handlers/update_task_status.dart (hunk 2 above)
Comment (liyuqian) [reply app_dart/lib/src/request_handlers/update_task_status.dart:136]: Done.

**Review by liyuqian — COMMENTED** [review]
Thanks for the review! Docs added :)

File: app_dart/lib/src/request_handlers/update_task_status.dart [hunk 3]
```diff
@@ -105,7 +105,12 @@ class UpdateTaskStatus extends ApiRequestHandler<UpdateTaskStatusResponse> {
       await _insertBigquery(commit, task);
     }
 
-    await _writeToMetricsCenter(resultData, scoreKeys, commit, task);
+    try {
+      await _writeToMetricsCenter(resultData, scoreKeys, commit, task);
+    } catch (e) {
```
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/update_task_status.dart:110]: From the log, the error is `DetailedApiRequestError`, do you think we can use the exact error?

File: app_dart/lib/src/request_handlers/update_task_status.dart [hunk 4]
```diff
@@ -105,7 +105,12 @@ class UpdateTaskStatus extends ApiRequestHandler<UpdateTaskStatusResponse> {
...
+      log.info(e.toString());
```
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/update_task_status.dart:111]: can we use log.error?

Comment (liyuqian) [reply app_dart/lib/src/request_handlers/update_task_status.dart]: Sure, done.

**Pull request merged**